File: samples/server/others/kotlin-server/jaxrs-spec/src/main/kotlin/org/openapitools/server/apis/DefaultApi.kt
package org.openapitools.server.apis;

import java.math.BigDecimal

import javax.ws.rs.*
import javax.ws.rs.core.Response


import java.io.InputStream



@Path("/")
@javax.annotation.Generated(value = arrayOf("org.openapitools.codegen.languages.KotlinServerCodegen"), comments = "Generator version: 7.10.0-SNAPSHOT")
class DefaultApi {

    @GET
    suspend fun findPetsByStatus(@PathParam("path_default") pathDefault: String,@PathParam("path_nullable") pathNullable: String,@QueryParam("query_default") @DefaultValue("available")   queryDefault: String,@QueryParam("query_default_enum") @DefaultValue("B")   queryDefaultEnum: String,@QueryParam("query_default_int") @DefaultValue("3")   queryDefaultInt: BigDecimal,@HeaderParam("header_default")  @DefaultValue("available")  headerDefault: String,@HeaderParam("header_default_enum")  @DefaultValue("B")  headerDefaultEnum: String,@HeaderParam("header_default_int")  @DefaultValue("3")  headerDefaultInt: BigDecimal,@CookieParam("cookie_default") @DefaultValue("available")  cookieDefault: String,@CookieParam("cookie_default_enum") @DefaultValue("B")  cookieDefaultEnum: String,@CookieParam("cookie_default_int") @DefaultValue("3")  cookieDefaultInt: BigDecimal,@QueryParam("query_nullable")   queryNullable: String?,@HeaderParam("header_nullable")   headerNullable: String?,@CookieParam("cookie_nullable")  cookieNullable: String?,@QueryParam("\$query-\$dollar-sign")   dollarQueryDollarDollarSign: String?): Response {
        return Response.ok().entity("magic!").build();
    }
}

File: samples/server/petstore/typescript-nestjs-server/builds/parameters/api/DefaultApi.ts
import { Injectable } from '@nestjs/common';
import { Observable } from 'rxjs';

export type FindPetsByStatusRequestParams = {
    pathDefault: string
    pathNullable: string
    queryDefault: string | undefined
    queryDefaultEnum: 'A' | 'B' | 'C' | undefined
    queryDefaultInt: number | undefined
    headerDefault: string | undefined
    headerDefaultEnum: 'A' | 'B' | 'C' | undefined
    headerDefaultInt: number | undefined
    cookieDefault: string | undefined
    cookieDefaultEnum: 'A' | 'B' | 'C' | undefined
    cookieDefaultInt: number | undefined
    queryNullable: string | null | undefined
    headerNullable: string | null | undefined
    cookieNullable: string | null | undefined
    $query$dollarSign: string | undefined
}

@Injectable()
export abstract class DefaultApi {
  abstract findPetsByStatus(findPetsByStatusRequestParams: FindPetsByStatusRequestParams, request: Request): void | Promise<void> | Observable<void>;


} 